tests/test_utils.py

```diff
@@ -16,10 +15,10 @@ class TestIn1D:
     @pytest.mark.skip_xp_backend(Backend.DASK_ARRAY, reason="no argsort")
     @pytest.mark.skip_xp_backend(Backend.SPARSE, reason="no unique_inverse, no device")
     # cover both code paths
-    @pytest.mark.parametrize("x2", [np.arange(9), np.arange(15)])
-    def test_no_invert_assume_unique(self, xp: ModuleType, x2: Array):
+    @pytest.mark.parametrize("n", [9, 15])
+    def test_no_invert_assume_unique(self, xp: ModuleType, n: int):
         x1 = xp.asarray([3, 8, 20])
-        x2 = xp.asarray(x2)
+        x2 = xp.arange(n)
```

Comment:
`NumPyReadOnly.asarray(x2, copy=None)` was tainting the globally defined x2, ruining it for the following iterations of the test.